lib/src/ui/pages/home/home_page.dart

```diff
@@ -33,51 +33,72 @@ class HomePage extends StatefulWidget {
 }
 
 class _HomePageState extends State<HomePage> {
-  final NYTimesRepository _nyTimesRepository = inject<NYTimesRepository>();
-  final ExceptionMapper _exceptionMapper = inject<ExceptionMapper>();
-  Future<List<Article>> _articleFuture;
+  final HomeNotifier _homeNotifier = inject<HomeNotifier>();
+
+  @override
+  void initState() {
+    super.initState();
+    _homeNotifier.fetchHomeArticles(context);
+  }
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       backgroundColor: Theme.of(context).backgroundColor,
       appBar: _buildAppBar(context),
       body: SafeArea(
-        child: Column(
-          children: [
-            Expanded(
-              child: SingleChildScrollView(
-                child: Column(
-                  crossAxisAlignment: CrossAxisAlignment.start,
-                  children: [
-                    const SizedBox(height: _Constants.sectionHeaderPadding),
-                    _buildNewsSectionHeader(
-                      title: Strings.of(context).mostViewed,
-                      routeBuilder: (context) => NewsListPage(
-                        headerTitle: Strings.of(context).mostViewed,
-                        listEntities: primaryNewsListEntities.toSecondaryNewsListEntities(),
-                      ),
-                    ),
-                    const SizedBox(height: _Constants.sectionHeaderPadding),
-                    PrimaryNewsListView(
-                      primaryNewsListEntities: primaryNewsListEntities.take(5).toList(),
-                    ),
-                    const SizedBox(height: _Constants.sectionSpacing),
-                    _buildNewsSectionHeader(
-                      title: Strings.of(context).newest,
-                      routeBuilder: (context) => NewsListPage(
-                        headerTitle: Strings.of(context).newest,
-                        listEntities: secondaryNewsListEntities,
-                      ),
-                    ),
-                    const SizedBox(height: _Constants.sectionHeaderPadding),
-                    _buildSecondaryNewsItems(),
-                  ],
-                ),
-              ),
-            ),
-          ],
-        ),
+        child: ChangeNotifierProvider.value(
+            value: _homeNotifier,
+            builder: (providerContext, child) {
+              final List<Article> mostPopularArticles =
+                  providerContext.select((HomeNotifier notifier) => notifier.mostPopularArticles);
+
+              final List<Article> newestArticles =
+                  providerContext.select((HomeNotifier notifier) => notifier.newestArticles);
+
+              final String headerTitle = getTitleForCriterion(providerContext);
+
+              return mostPopularArticles == null
```

Comment:
brakuje obsługi błędu

Reply:
wolałbym, żeby obsługa błędów to był osobny task